src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -10,502 +11,314 @@
+            {
+                get => J;
+                set => J = (TNested)value;
+            }
 
-        private class TestClassWithParamCtor<TNested> : TestClass<TNested>
-        {
-            public TestClassWithParamCtor(string b, int c, int? d, float e, float? f, bool g, bool? h, int[] i)
-                => (B, C, D, E, F, G, H, I) = (b, c, d, e, f, g, h, i);
+            void ITestObject.Initialize(INestedObject nested)
+            {
+                A = "Hello";
+                B = null;
+                C = 42;
+                D = null;
+                E = 3.14e+17f;
+                G = true;
+                I = new int[] { 42, 17 };
+                nested.Initialize();
+                J = (TNested)nested;
+            }
+
+            void ITestObject.Verify()
+            {
+                Assert.Equal("Hello", A);
+                Assert.Null(B);
+                Assert.Equal(42, C);
+                Assert.Null(D);
+                Assert.Equal(3.14e17f, E);
+                Assert.True(G);
+                Assert.Collection(I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));
+                Assert.NotNull(J);
+                J.Verify();
+            }
         }
 
-        private class TestValueType<TNested>
+        private class TestValueType<TNested> : ITestObject where TNested : INestedObject
         {
             public string A { get; set; }
             public string B { get; set; }
             public int C { get; set; }
             public int? D { get; set; }
             public float E { get; set; }
-            public float? F { get; set; }
             public bool G { get; set; }
-            public bool? H { get; set; }
             public int[] I { get; set; }
             public TNested J { get; set; }
-        }
+            INestedObject ITestObject.J
+            {
+                get => J;
+                set => J = (TNested)value;
+            }
 
-        private class NestedClass
-        {
-            public string A { get; set; }
-            public int B { get; set; }
-        }
+            void ITestObject.Initialize(INestedObject nested)
+            {
+                A = "Hello";
+                B = null;
+                C = 42;
+                D = null;
+                E = 3.14e+17f;
+                G = true;
+                I = new int[] { 42, 17 };
+                nested.Initialize();
+                J = (TNested)nested;
+            }
 
-        private class NestedClassWithParamCtor : NestedClass
-        {
-            public NestedClassWithParamCtor(string a)
-                => A = a;
+            void ITestObject.Verify()
+            {
+                Assert.Equal("Hello", A);
+                Assert.Null(B);
+                Assert.Equal(42, C);
+                Assert.Null(D);
+                Assert.Equal(3.14e17f, E);
+                Assert.True(G);
+                Assert.Collection(I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));
```

Comment:
nit: I thought there was a simpler way to compare arrays rather than this element-by-element lambda. Would something like `Assert.Equal(output_array, expected_array)` work?

Reply:
`Assert.True(I.SequenceEqual(new[] { 42, 17 }));` would work but the message is not very expressive when it fails

Reply:
I changed it